### PR TITLE
[BD-46] docs: add auto-pulling theme variables to components pages

### PR DESCRIPTION
**Pull request by viktorrusakov**
- auto-pull theme variables to components pages when creating the pages
- update local development documentation and fix couple of broken links

JIRA: [PAR-364](https://openedx.atlassian.net/browse/PAR-364)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @viktorrusakov! I've created [BLENDED-1042](https://openedx.atlassian.net/browse/BLENDED-1042) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

**Comment by viktorrusakov** [comment]
Note that on [FormCheckbox component's page](https://deploy-preview-931--paragon-edx.netlify.app/components/form/form-checkbox/) all variables related to the `Form` component are shown, not just the ones related to `FormCheckbox`, the same goes for all `Form` subcomponents pages. 

This is due to the code structure of `Form` components: all of them are located inside root directory of `Form` component and they share same `_variables.scss` file. We might need to refactor this so that only variables directly related to the components are shown.

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ See the [code of conduct](https://github.com/edx/.github/blob/master/CODE_OF_CON
 
 #### 1. Start the documentation site development server
 
-The Paragon documentation site serves both as documentation and as a workbench to create your component within. To see your component in action, you need to run the documentation site locally. (Note you need to install dependences both in the project root and the `www` directory)
+The Paragon documentation site serves both as documentation and as a workbench to create your component within. To see your component in action, you need to run the documentation site locally. (Note you need to install dependencies both in the project root and the `www` directory)
```
Comment (adamstankiewicz) [inline README.md:56]: 🔥

**Review by adamstankiewicz — APPROVED** [review]
Awesome! This is fantastic. 💯

**Comment by edx-webhook** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/paragon&entry.867055334=https://github.com/edx/paragon/pull/931&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2021-12-08+15:46&entry.1917517419=2021-12-10+11:51&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**